webserver.h

```diff
@@ -73,27 +86,55 @@ class WebServer
         CROW_ROUTE(_crowApp, "/api/sockets/<int>") 
             .methods(crow::HTTPMethod::GET)([&](int socketId) -> crow::response
             {
-                return nlohmann::json(_controller.GetSocketById(socketId)).dump();
+                try
+                {
+                    shared_lock readLock(_apiMutex);
+                    return nlohmann::json{{"socket", _controller.GetSocketById(socketId)}}.dump();
+                }
+                catch(const std::exception& e)
+                {
+                    logger->error("Error in /api/sockets/{}: {}", socketId, e.what());
+                    return {crow::BAD_REQUEST, string("Error: ") + e.what()};
+                }
             });
 
         // Enumerate all the canvases
 
         CROW_ROUTE(_crowApp, "/api/canvases")
             .methods(crow::HTTPMethod::GET)([&]() -> crow::response
             {
-                return nlohmann::json(_controller.Canvases()).dump();
+                try
+                {
+                    shared_lock readLock(_apiMutex);
+                    return nlohmann::json(_controller.Canvases()).dump();
+                }
+                catch(const std::exception& e)
+                {
+                    logger->error("Error in /api/canvases: {}", e.what());
+                    return {crow::BAD_REQUEST, string("Error: ") + e.what()};
+                }
             });
 
         // Detail a single canvas
 
         CROW_ROUTE(_crowApp, "/api/canvases/<int>")
             .methods(crow::HTTPMethod::GET)([&](int id) -> crow::response
             {
-                auto allCanvases = _controller.Canvases();
-                if (id < 0 || id >= allCanvases.size())
-                    return {crow::NOT_FOUND, R"({"error": "Canvas not found"})"};
-
-                return nlohmann::json(allCanvases[id]).dump(); 
+                try
+                {
+                    shared_lock readLock(_apiMutex);
```

Comment:
There was a bad assumption about the ID always being less than the vector size, which isn't true after deletion/insertion